pkg/sql/gcjob_test/gcjob_test_test: TestGCJobRetry failed

pkg/sql/gcjob_test/gcjob_test_test.TestGCJobRetry [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/12065779?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/12065779?buildTab=artifacts#/) on master @ [767e1353dd817fb5bea459c0b729ef23988ccad6](https://github.com/cockroachdb/cockroach/commits/767e1353dd817fb5bea459c0b729ef23988ccad6):


```
=== RUN   TestGCJobRetry
    test_log_scope.go:170: test logs captured to: /artifacts/tmp/_tmp/4d64db87b2aa0e9fccbce189bc4b8745/logTestGCJobRetry2866189077
    test_log_scope.go:81: use -show-logs to present logs inline
    test_server_shim.go:111: cluster virtualization disabled in global scope due to issue: #76378 (expected label: C-bug)
    gc_job_test.go:327: error scanning '&{<nil> 0xc0059c6700}': sql: Scan error on column index 3, name "last_run": converting NULL to string is unsupported
    panic.go:522: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/4d64db87b2aa0e9fccbce189bc4b8745/logTestGCJobRetry2866189077
--- FAIL: TestGCJobRetry (4.13s)
```
<p>Parameters: <code>TAGS=bazel,gss</code>
, <code>stress=true</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-foundations
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestGCJobRetry.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-32089

--- a/pkg/sql/gcjob_test/gc_job_test.go
+++ b/pkg/sql/gcjob_test/gc_job_test.go
@@ -12,6 +12,7 @@ package gcjob_test
 
 import (
 	"context"
+	gosql "database/sql"
 	"fmt"
 	"strconv"
 	"sync/atomic"
@@ -320,18 +321,22 @@ SELECT job_id
 
 	const expectedRunningStatus = string(sql.RunningStatusWaitingForMVCCGC)
 	testutils.SucceedsSoon(t, func() error {
-		var status, runningStatus, lastRun, nextRun, numRuns, jobErr string
+		var status, runningStatus, lastRun, nextRun, numRuns, jobErr gosql.NullString
 		tdb.QueryRow(t, fmt.Sprintf(`
 SELECT status, running_status, error, last_run, next_run, num_runs
 FROM crdb_internal.jobs
 WHERE job_id = %s`, jobID)).Scan(&status, &runningStatus, &jobErr, &lastRun, &nextRun, &numRuns)
 
-		t.Logf(`details about SCHEMA CHANGE GC job: {status: %q, running_status: %q, error: %q, last_run: %q, next_run: %q, num_runs: %q}`,
+		t.Logf(`details about SCHEMA CHANGE GC job: {status: %#v, running_status: %#v, error: %#v, last_run: %#v, next_run: %#v, num_runs: %#v}`,
 			status, runningStatus, jobErr, lastRun, nextRun, numRuns)
 
-		if runningStatus != expectedRunningStatus {
-			return errors.Newf(`running_status %s does not match expected status %s`,
-				runningStatus, expectedRunningStatus)
+		if !runningStatus.Valid {
+			return errors.Newf(`running_status is NULL but expected %q`, expectedRunningStatus)
+		}
+
+		if actualRunningStatus := runningStatus.String; actualRunningStatus != expectedRunningStatus {
+			return errors.Newf(`running_status %q does not match expected status %q`,
+				actualRunningStatus, expectedRunningStatus)
 		}
 
 		return nil

